KFSPTS-26275 Add CNCR Request V4 workflow handling

cah292 opened the pull request:
This is an updated version of the declined #1376 changes regarding the new Concur API workflow handling. As was discussed in the user story, the new version of this code will always approve incoming Travel Requests, even if their accounts fail validation.

In addition, this PR will add an extra approve-anyway warning to the validation messages if a Request fails account validation. The message is not being added to the workflow action itself, but it will be added to the Event Notification V2 report.

Please let me know if you have any questions or concerns about this updated approach.

jhulslander reviewed (commented):
changes look good, I noticed one thing that could be adjusted

cah292 commented on src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestV4ServiceImpl.java:
```diff
@@ -224,10 +238,106 @@ protected void updateRequestStatusInConcur(
+            return;
+        }
+        
+        String requestId = resultsDTO.getReportNumber();
```
Refactoring.

cah292 commented:
I've refactored the code for clarity as requested. I also ran the related unit tests locally (and adjusted them as needed) to ensure that things still work as expected.

This is ready for review again.

jhulslander reviewed (approved):
changes look good, unit tests pass, I shall merge
